### ISSUE TITLE
Comments interfering with fluent formatting

### ISSUE THREAD

**Issue by max-sixty** [ambv/black#458]
Copied over from https://github.com/ambv/black/issues/379#issuecomment-409710377

Without a comment, formatted with black (`-l 79`)

```python
    return (
        df_day.assign(date=date, fund_ticker=ticker)
        .pipe(lambda x: x.loc[x["Shares"].notnull()])
        .assign(Shares=df_day["Shares"].astype("float64"))
    )
```

Adding a comment, before formatting

```python
    return (
        df_day.assign(date=date, fund_ticker=ticker)
        .pipe(lambda x: x.loc[x["Shares"].notnull()])
        # arrives as int sometimes
        .assign(Shares=df_day["Shares"].astype("float64"))
    )
```

With a comment, after formatting with black:

```python
    return (
        df_day.assign(date=date, fund_ticker=ticker).pipe(
            lambda x: x.loc[x["Shares"].notnull()]
        )
        # arrives as int sometimes
        .assign(Shares=df_day["Shares"].astype("float64"))
    )
```

Without thinking about the more general problem, the second example here is ideal - the comment can be _within_ a fluent expression and wouldn't change the fluent expression formatting.

Thanks again for such an awesome library and vision!

Operating system: MacOS
Python version: 3.7
*Black* version: 18.6b4
Does also happen on master:

**Comment by ambv** [comment]
I agree with you that comments should not break the flow of fluent interfaces. We'll address it next week.

**Comment by ambv** [comment]
The problem is that the standalone comment "splits" the fluent interface so Black sees two periods before (and fluent interfaces are implemented to only apply for >2 periods), then a comment, and one after.

**Comment by ambv** [comment]
See [this Black playground](https://black.now.sh/?version=stable&state=_Td6WFoAAATm1rRGAgAhARYAAAB0L-Wj4AJiAMddAD2IimZxl1N_WmXWfJWiTYL4Me9qLRkrw2xxhVHtniklAbo147Ca7kgobVhG9wydJsPAHf9bhpVJbxamy0ybcLVX3RfphVICosau3mNLd_4ahwFVp9NyRHDfQYAXpsKDtn30K0dlX2sJIrbxbWabrtSMvG-fob4w_Xt3ia_t312KJiu2J_4HH6PP2AcLJvLRHWa_jbE_5MfHzBHepENkrv5pP--QtB3fT7E2z75CJggHronFOUViZ_ljt5m7mZsTcZKvs0hlsTkAAG7-k8iGf8quAAHjAeMEAABL0iatscRn-wIAAAAABFla).
